7_react/react1/week2/react-app/src/components/List.js

```diff
@@ -0,0 +1,88 @@
+import React, { useEffect, useState } from "react";
+import { ListItem } from "./ListItem";
+
+export function List() {
+  const [list, setList] = useState(initialList);
+
+  useEffect(() => {
+    console.log("something");
+  }, [list]); // dependency on list changes
+
+  useEffect(() => {
+    console.log("hello");
+  }, []); // empty dependency array = only when the component mounts;
+
+  const addRandomTask = () => {
+    const newTask = {
+      id: list.length + 1,
+      description: `Random task ${list.length + 1}`,
+      done: false,
+    };
+    setList((previousState) => [...previousState, newTask]);
+  };
+
+  const markAsDone = (id) => {
+    setList((previousState) =>
+      previousState.map((task) =>
+        task.id === id ? { ...task, done: !task.done } : task
+      )
+    );
+  };
+
+  const deleteTask = (id) => {
+    setList((previousState) => previousState.filter((task) => task.id !== id));
+  };
+
+  return (
+    <ul>
+      <WatchCount />
+      <button onClick={addRandomTask}>Add random item</button>
+      {list.length === 0 ? (
+        <div>No items...</div>
+      ) : (
+        list.map((listItem) => (
+          <ListItem
+            key={listItem.id}
+            id={listItem.id}
+            description={listItem.description}
+            done={listItem.done || false}
+            markAsDone={markAsDone}
+            deleteTask={deleteTask}
+          />
+        ))
+      )}
+    </ul>
+  );
+}
```

Comment:
Very nice separation into different components. The code is easy to read. Good job remembering to set the key.